test/integration/targets/openssh_keypair/tests/validate.yml

```
@@ -37,3 +37,14 @@
   assert:
     that:
       - privatekey4.stat.exists == False
+
+
+- name: Validate privatekey5 (test - Public key output)
+  debug:
+    msg: "{{ lookup('file', output_dir ~ '/privatekey5.pub').strip('\n') }}"
+  register: privatekey5
+
+- name: Validate privatekey5 (assert - Public key module output equal to the public key on host)
+  assert:
+    that:
+      - publickey_gen.public_key == privatekey5.msg
```

You can also do this in one task:
```suggestion
      - "publickey_gen.public_key == lookup('file', output_dir ~ '/privatekey5.pub').strip('\n')"
```